File: internal/sast/fixtures/python/nosql_negative.py
# internal/sast/fixtures/python/nosql_negative.py
from flask import Flask, request, jsonify
from pymongo import MongoClient

app = Flask(__name__)
db = MongoClient()["app"]

@app.route("/login", methods=["POST"])
def login():
    username = str(request.json.get("username", ""))
    password = str(request.json.get("password", ""))
    user = db.users.find_one({"username": username, "password": password})
    return jsonify(user is not None)

File: internal/sast/fixtures/javascript/nosql_negative.js
// internal/sast/fixtures/javascript/nosql_negative.js
const express = require("express");
const User = require("./models/user");
const app = express();

app.post("/login", async (req, res) => {
  const username = String(req.body.username || "");
  const password = String(req.body.password || "");
  const user = await User.findOne({ username, password });
  res.json({ success: !!user });
});
